src/common/card/Card.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import UserIcon from "../buttons/UserIcon";
+import XButton from "../buttons/XButton";
+import SendButton from "../buttons/SendButton";
+import SaveButton from "../buttons/SaveButton";
+import './Card.css';
+
+// props: 
+//      All: type, user, hasX, topText, mainText, hasLine, bottomText
+//      Post-specific (complex): bookmarked, show, handleBookmark, handleShow (if hasX), isProfileList, isUserSavedList
+//      Ex. Contact (simple): no additional props
+
+const ACard = React.memo((props) => {
+    return (
+        ((props.type !== 'complex' || 
+            ((!props.isUserSavedList && props.isProfileList) || 
+            (props.isUserSavedList && props.bookmarked) || 
+            (!props.isProfileList && props.show))) &&
+         <div className='card'>{props.children}</div>
+        )
+    )
+})
+const CardBody = (props) => {
+    return (
+        <div className='colContainer'>{props.children}</div>
+    )
+}
+const X = (props) => {
+    return (
+        <div>{props.hasX && <XButton handleShow={props.handleShow}/>}</div>
+    )
+}
+const CardInfo = (props) => {
+    return (
+        <div className='cardTextBlock'>{props.children}</div>
+    )
+}
+const Name = (props) => {
+    return (
+        <p>{props.topText}</p>
+    )
+}
+const MainText = (props) => {
+    return (
+        <p className='mainText'>{props.mainText}</p>
+    )
+}
+const Line = (props) => {
+    return (
+        <div>
+            {props.hasLine && <div className='postLine'/>}
+        </div>
+    )
+}
+const BottomText = (props) => {
+    return (
+        <p style={{whiteSpace: 'pre'}}>
+            {props.bottomText}
+        </p>
+    )
+}
+const SendSave = (props) => {
+    return (
+        <div>
+            {props.type === 'complex' &&
+            <div className='sendSaveContainer'>
+                <SendButton/>
+                <SaveButton bookmarked={props.bookmarked}
+                            handleBookmark={props.handleBookmark}/>
+            </div>}
+        </div>
+    )
+}
+const Card = React.memo((props) => {
+    return (
+        <ACard type={props.type} isProfileList={props.isProfileList} isUserSavedList={props.isUserSavedList} bookmarked={props.bookmarked} show={props.show}>
```

Comment:
It does not makes sense to use `React.memo` on `ACard` in this case because you are passing `children` to `ACard`.
Every time `Card` rerenders, all the children you are passing to `ACard` (`UserIcon`, `CardBody`, etc) get rebuilt with new references. This means `ACard` will rerender every single time `Card` rerenders. Even if `Card` is rerendering due to a changed prop that isn't being passed to`ACard`.